homeassistant/components/magichome/__init__.py

```diff
@@ -0,0 +1,160 @@
+"""Support for MagicHome Smart devices."""
+from datetime import timedelta
+import logging
+import voluptuous as vol
+
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, CONF_PLATFORM
+from homeassistant.helpers import discovery
+from homeassistant.helpers.dispatcher import dispatcher_send, async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_COMPANY = "company"
+
+DOMAIN = "magichome"
+DATA_MAGICHOME = "data_magichome"
+
+SIGNAL_DELETE_ENTITY = "magichome_delete"
+SIGNAL_UPDATE_ENTITY = "magichome_update"
+
+SERVICE_FORCE_UPDATE = "force_update"
+SERVICE_PULL_DEVICES = "pull_devices"
+
+MAGICHOME_TYPE_TO_HA = {
+    "light": "light",
+    "scene": "scene",
+    "switch": "switch",
+    "socket": "socket",
+    "controller": "controller",
+}
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(CONF_COMPANY, default="ZG001"): cv.string,
+                vol.Optional(CONF_PLATFORM, default="ZG001"): cv.string,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def setup(hass, config):
+    """Set up MagicHome Component."""
+    from magichome import MagicHomeApi
```

Comment:
Import should be placed on the top of the file.

Reply:
OK！